schema/tx_schema.md

```diff
@@ -0,0 +1,155 @@
+## Overview
+
+Signal size can be automatically calculated or explicitly specified. Cannot (currently) specify signal position.
+
+## Default TX message definition fields
+
+```json
+{
+    "message_name": {
+        "signals": {
+            "signal_name": {
+                "resolution": 1,
+                "offset": 0,
+                "signed": false,
+                "min": 0
+            }
+        }
+    }
+}
+```
+
+## Absolute minimum message definitions
+
+Given the default fields there are two options of absolute minimum TX message definitions:
+```json
+{
+    "message_name": {
+        "message_id": 123,
+        "signals": {
+            "signal_name_1": {
+                "bits": 14
+            },
+            "signal_name_2": {
+                "max": 100
+            }
+        }
+    }
+}
+```
+
+## Associate an enum definition with a signal
+
+The required signal size/sign is automatically determined, even if the enum is negative. It's illegal to override this by defining `min`, `max` and/or `signed`.
+
+Note that this is legal:
+```json
+{
+    "message_name": {
+        "signals": {
+            "signal_name_1": {
+                "enum": "example_enum_1"
+            }
+        }
+    }
+}
+```
+
+Whereas this is illegal, even if it correctly describes the bounds of the enum:
+```json
+{
+    "message_name": {
+        "signals": {
+            "signal_name_2": {
+                "enum": "example_enum_2",
+                "min": -3,
+                "max": 3,
+                "signed": true

```

Comment:
Do we need to specify that the data is signed? For cases like this it could be smart to autogenerate the offset to -3 and not have to deal with signed number

Reply:
nope, but i was illustrating this point
> It's illegal to override this by defining `min`, `max` and/or `signed`.

Reply:
removing signed field - ignore